packages/supercolliderjs/README.md

```diff
@@ -0,0 +1,134 @@
+# supercollider.js
+
+[![NPM downloads][npm-downloads-image]][npm-url] [![MIT License][license-image]][license-url] [![Dependency Status](https://david-dm.org/crucialfelix/supercolliderjs.svg)](https://david-dm.org/crucialfelix/supercolliderjs)
+
+The JavaScript client library for SuperCollider.
+
+SuperCollider is an environment and programming language for real time audio synthesis and algorithmic composition. It provides an interpreted object-oriented language which functions as a network client to a state of the art, realtime sound synthesis server.
+
+This library provides functionality for working with:
+
+- scsynth (the synthesis server)
+- sclang (supercollider language interpreter)
+
+
+## Documentation
+
+[API](https://crucialfelix.github.io/supercolliderjs/api/)
+[Guide](https://crucialfelix.gitbooks.io/supercollider-js-guide/content/)
+
+## Features
+
+- Send and receive OSC messages
+- Comprehensive support for calling all commands the server understands
+- Call async commands on the server receive results as Promises
+- Synth/Group/Bus/Buffer allocators with clean immutable state implementation
+- Server state and synth/group tracking
+- Just-in-time OSC scheduler
+- Codebase written with Flow (type checking)
+- High unit test coverage
+
+- Dryadic: declarative DSL for managing component trees.
+
+## Examples
+
+See also the [Examples Repository](https://github.com/crucialfelix/supercolliderjs-examples).
+
+### language
+
+Interpret SuperCollider language code.
+
+```javascript
+var sc = require('supercolliderjs');
+
+sc.lang.boot().then(function(sclang) {
+
+  sclang.interpret('(1..8).pyramid')
+    .then(function(result) {
+      // result is a native javascript array
+      console.log('= ' + result);
+    }, function(error) {
+      // syntax or runtime errors
+      // are returned as javascript objects
+      console.error(error);
+    });
+
+}, function(error) {
+  console.error(error)
+  // sclang failed to startup:
+  // - executable may be missing
+  // - class library may have failed with compile errors
+});
+```
+
+### server
+
+```javascript
+let sc = require('supercolliderjs');
+
+sc.server.boot().then((server) => {
```

Comment:
not important, but a little idiomatic thing i'm pondering about:

is see some places we refer to a `server`, a `scsynth` and a `lang` and a `sclang`. are these two pairs of concept logically the same in supercollider or is there a reason we use `server` here instead of `scsynth` as a variable name? 

i see `Lang` and `Server` used in the new api, but i see somewhere `SCLang` also used:

```js
import SCLang from "@supercollider.js/lang";
```
 should we then maybe do

```js
import SCSynth from "@supercollider.js/server";
```
?

Reply:
> not important

Oh I think it is important. I agree and have been trying to clean this up.

- In supercollider there is a `Server` class. It's roughly similar in purpose to supercollider.js `Server`. (controls and communicates with scsynth process)
- The binary executable is called `scsynth`.
- A `Synth` is a single node of synthesis with many `UGen` and inputs and outputs for patching to other Synths.  (legacy confusion)

Although `Server` is a bit generic sounding in node.js, I figure it's best to use the same name as the class of similar functionality in supercollider.

- The `SCLang` class is similar to`Server` in that it controls and communicates with the `sclang` process.
- The language binary is called `sclang`

I could call it `Lang` but that seems very generic and doesn't add meaning.

I named the packages:
- @supercollider.js/lang
- @supercollider.js/server

I intend to add packages:
- @supercollider.js/sclang
- @supercollider.js/scsynth

which would include the binaries and class files and everything. It would then be a zero config, portable, working supercollider. By goal is to make supercollider a truly network native , portable synthesis engine. This is what James envisioned when he embraced OSC.

Reply:
> By goal is to make supercollider a truly network native , portable synthesis engine. This is what James envisioned when he embraced OSC.

thumbs up!

Reply:
Renamed to `@supercollider/`